migration/v2/model/application.go

```diff
@@ -105,3 +108,24 @@ type ImportTag struct {
 	ImportID uint `gorm:"index"`
 	Import   *Import
 }
+
+//
+// Validation Hook to avoid cyclic dependencies.
+func (dep *Dependency) BeforeCreate(db *gorm.DB) (err error) {
```

Comment:
The GORM Hook on the model level should cover both API&CSV import use cases.

Reply:
What about update _"Manage Dependencies"_ use cases?